### DISCUSSION TITLE
Recommended way to manage open connection

### DISCUSSION THREAD

**Discussion by pmbanugo** [kriszyp/lmdb-js#301] [Q&A] [answered]
Hi, what's the recommended way to manage open connections? Do I keep one `Database` instance via a Singleton and keep the connection open for the lifetime of that process, or keep creating & closing connections? I'm using it as a sort of cache and I don't know how expensive it is to create many open connection (using `open()`) based on the number of requests.

Also, when a path isn't specified for `open()`, and the application exits without calling `db.close()`, will the temp database still be deleted? Maybe this is internally handled by lmdb-js or I should do that.

**Comment by kriszyp** [comment, marked as answer]
Yes, the recommended approach is to use a singleton database instance, and let all db consumers interact with that singleton. lmdb-js is in-process, so it is different the database architecture that involves a network connection, whereas lmdb-js instance doesn't have any type of network connection, it is just an instance with the state of the interface to the database (and there isn't any connection pipe that can get blocked).

> Also, when a path isn't specified for open(), and the application exits without calling db.close(), will the temp database still be deleted? 

Yes, the temp database will be deleted as long as the process is terminated through normal means. But `kill -9`, for example, would immediately terminate, without any cleanup. The temp database is also placed the OS's temp directory path, which traditionally cleaned up at some point, if there is an abnormal exit.

**Reply by pmbanugo** [reply]
Thank you!
